great_expectations/experimental/datasources/type_lookup.py

```diff
@@ -67,6 +67,9 @@ def __delitem__(self, key: ValidTypes):
         super().pop(value, None)
 
     def __setitem__(self, key: ValidTypes, value: ValidTypes):
+        if key in self and value in self and self[key] == value and self[value] == key:
```

Comment:
Subclasses try to register the associated asset types in the superclass. I allow this and return early when this is done.